docs/src/main/asciidoc/config-custom.adoc

```diff
@@ -0,0 +1,112 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Custom Configuration
+
+include::./attributes.adoc[]
+
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:toc:
+
+[[custom-config-source]]
+== Custom Config Sources
+
+It is possible to create a Custom Config Sources as specified in
+link:https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#custom-configsources-via-configsourceprovider[MicroProfile Config Source].
+
+A custom Config Source requires an implementation of `org.eclipse.microprofile.config.spi.ConfigSource` or
+`org.eclipse.microprofile.config.spi.ConfigSourceProvider`. Each implementation requires registration via
+the https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html[ServiceLoader] mechanism, either in
+`META-INF/services/org.eclipse.microprofile.config.spi.ConfigSource` or
+`META-INF/services/org.eclipse.microprofile.config.spi.ConfigSourceProvider`.
+
```

Comment:
Should we add a very simple example? 
I found this a bit short for something that is a recurring question.